libs/utils/src/lib/services/broadcast-channel/broadcast-channel.service.ts

```diff
@@ -0,0 +1,95 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { EMPTY, fromEvent, Observable } from 'rxjs';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class BroadcastChannelService {
+	private broadcastChannel: Record<string, BroadcastChannel> = {};
+
+	constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+	/**
+	 * initChannel
+	 *
+	 * The initChannel method initializes a new BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 */
+	public initChannel(channelName: string): void {
+		if (!channelName) {
+			console.warn('channelName is required');
+
+			return;
+		}
+
+		if (isPlatformBrowser(this.platformId) && !this.broadcastChannel[channelName]) {
```

Comment:
Can we use the WindowService's `runInBrowser` function here?

Reply:
I've not used it to keep the service something that can stand by itself, but I could switch to the service as well if there is a preference to do so.